tests/components/vera/test_config_flow.py

```diff
@@ -0,0 +1,187 @@
+"""Vera tests."""
+from unittest.mock import MagicMock
+
+from mock import patch
+from requests.exceptions import RequestException
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.vera import CONF_CONTROLLER, DOMAIN
+from homeassistant.components.vera.config_flow import VeraFlowHandler
+from homeassistant.const import CONF_EXCLUDE, CONF_LIGHTS, CONF_SOURCE
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_aync_step_user_success(hass: HomeAssistant) -> None:
+    """Test function."""
+    with patch("pyvera.VeraController") as vera_controller_class_mock:
+        controller = MagicMock()
+        controller.refresh_data = MagicMock()
+        controller.serial_number = "serial_number_0"
+        vera_controller_class_mock.return_value = controller
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result.get("type") == RESULT_TYPE_FORM
+        assert result.get("step_id") == config_entries.SOURCE_USER
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={
+                CONF_CONTROLLER: "http://127.0.0.1:123/",
+                CONF_LIGHTS: "12 13",
+                CONF_EXCLUDE: "14 15",
+            },
+        )
+        assert result.get("type") == RESULT_TYPE_CREATE_ENTRY
```

Comment:
We should use `dict[key]` when we expect the key to be in the dict.
```suggestion
        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
```